ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -828,38 +716,7 @@ static int
 S3FlushFinish(WT_STORAGE_SOURCE *storage, WT_SESSION *session, WT_FILE_SYSTEM *fileSystem,
   const char *source, const char *object, const char *config)
 {
-    S3Storage *s3 = (S3Storage *)storage;
-    S3FileSystem *fs = (S3FileSystem *)fileSystem;
-    // Constructing the pathname for source and cache from file system and local.
-    std::string srcPath = S3SourcePath(fs->homeDir, source);
-    std::string destPath = S3CachePath(fs->cacheDir, object);
-
-    // Converting S3 object name to cache directory structure to link the cache file with local
-    // file.
-    std::filesystem::create_directories(std::filesystem::path(destPath).parent_path());
-
-    // Linking file with the local file.
-    std::error_code ec;
-    std::filesystem::create_hard_link(srcPath.c_str(), destPath.c_str(), ec);
-    int ret = ec.value();
-    if (ret != 0) {
-        ret = errno;
-        s3->log->LogErrorMessage(
-          "S3FlushFinish: link " + std::string(source) + " to " + destPath + " failed");
-        return (ret);
-    }
-
-    // The file should be read-only.
-    std::filesystem::permissions(destPath.c_str(),
-      std::filesystem::perms::owner_read | std::filesystem::perms::group_read |
-        std::filesystem::perms::others_read,
-      std::filesystem::perm_options::add, ec);
-    ret = ec.value();
-    if (ret != 0) {
-        ret = errno;
-        s3->log->LogErrorMessage("S3FlushFinish: read permissions of " + destPath + " failed");
-    }
-    return (ret);
```

Comment:
I think I'm missing something - was this all for the caching subsystem? It's a little surprising to me that this function can now be a no-op.

Reply:
Ah yes, overall, throughout the tiered storage changes, there is a realisation that a flush finish might not be needed at all. It was meant to be called post flushing, but there is nothing really to do.

Reply:
OK, that makes sense. I assume we still need to provide something here since it's a callback? If so, maybe we could have a ticket to remove that callback from the API. Doing that removal feels out of scope for this ticket.

Reply:
Yes, I believe this has been mentioned already when finishing the azure and gcp project.